fastparquet/writer.py

```diff
@@ -747,15 +800,116 @@ def partition_on_columns(data, columns, root_path, partname, fmd, sep,
             fmd.row_groups.append(rg)
 
 
-def write_common_metadata(fn, fmd, open_with=default_openw):
-    """For hive-style parquet, write schema in special shared file"""
-    if isinstance(fn, str):
-        f = open_with(fn)
-    else:
-        f = fn
-    f.write(MARKER)
-    fmd.row_groups = []
-    foot_size = write_thrift(f, fmd)
-    f.write(struct.pack(b"<i", foot_size))
-    f.write(MARKER)
-    f.close()
+def write_common_metadata(fn, fmd, open_with=default_open,
+                          no_row_groups=True):
+    """
+    For hive-style parquet, write schema in special shared file
+
+    Parameters
+    ----------
+    fn: str
+        Filename to write to
+    fmd: thrift FileMetaData
+        Information to write
+    open_with: func
+        To use to create writable file as f(path, mode)
+    no_row_groups: bool (True)
+        Strip out row groups from metadata before writing - used for "common
+        metadata" files, containing only the schema.
+    """
+    with open_with(fn, 'wb') as f:
+        f.write(MARKER)
+        if no_row_groups:
+            rgs = fmd.row_groups
+            fmd.row_groups = []
+            foot_size = write_thrift(f, fmd)
+            fmd.row_groups = rgs
+        else:
+            foot_size = write_thrift(f, fmd)
+        f.write(struct.pack(b"<i", foot_size))
+        f.write(MARKER)
+
+
+def merge(file_list, verify_schema=True, open_with=default_open):
+    """
+    Create a logical data-set out of multiple parquet files.
+
+    The files referenced in file_list must either be in the same directory,
+    or at the same level within a structured directory, where the directories
+    give partitioning information. The schemas of the files should also be
+    consistent.
+
+    Parameters
+    ----------
+    file_list: list of paths
+    verify_schema: bool (True)
+        If True, will first check that all the schemas in the input files are
+        identical.
+    open_with: func
+        Used for opening a file for writing as f(path, mode)
```

Comment:
Oh, I see that this takes filenames rather than ParquetFiles.  Is there a reason to prefer one over the other?  My first expectation was that this would consume and produce ParquetFile objects.